PersonApi.Tests/V1/Controllers/PersonApiControllerTests.cs

```diff
@@ -0,0 +1,63 @@
+using AutoFixture;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NUnit.Framework;
+using PersonApi.V1.Boundary.Response;
+using PersonApi.V1.Controllers;
+using PersonApi.V1.UseCase.Interfaces;
+using System;
+using System.Threading.Tasks;
+
+namespace PersonApi.Tests.V1.Controllers
+{
+    [TestFixture]
+    public class PersonApiControllerTests
+    {
+        private Mock<IGetByIdUseCase> _mockGetByIdUseCase;
+        private PersonApiController _sut;
+        private readonly Fixture _fixture = new Fixture();
+
+        [SetUp]
+        public void Setup()
+        {
+            _mockGetByIdUseCase = new Mock<IGetByIdUseCase>();
+            _sut = new PersonApiController(_mockGetByIdUseCase.Object);
+        }
+
+        [Test]
+        public async Task GetPersonByIdAsyncNotFoundReturnsNotFound()
+        {
+            var id = Guid.NewGuid();
+            _mockGetByIdUseCase.Setup(x => x.ExecuteAsync(id)).ReturnsAsync((PersonResponseObject) null);
+            var response = await _sut.GetPersonByIdAsync(id).ConfigureAwait(false);
+
+            response.Should().BeOfType(typeof(NotFoundObjectResult));
+            (response as NotFoundObjectResult).Value.Should().Be(id);
```

Comment:
Can we also add checks for what status codes we are expecting in those different scenarios?

Reply:
That is already done in this line - the check is implied in the object type.
The `NotFoundObjectResult` type has a fixed status code of 404. The `OkObjectResult` type has a fixed status code of 200.